libazureinit/src/user.rs

```diff
@@ -50,6 +50,18 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
+pub fn add_user_for_passwordless_sudo(
+    username: &str,
+) -> Result<(), Error>{
+    let mut sudoers_file = std::fs::OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open("/etc/sudoers")?;
```

Comment:
The distributions I'm familiar with (I checked Gentoo, Debian, Ubuntu, Fedora, and RHEL) configure sudo to read additional configuration snippets from `/etc/sudoers.d`.

This will include any file in the `/etc/sudoers.d/` directory as configuration which makes it easy to drop in additional configuration. This makes it easy to drop in snippets of config without parsing the main config to make sure it's not already there, and also when the distribution configuration is updated it doesn't potentially zap your config change (depending on how administrators handle new config files from package updates).

I think we should certainly _prefer_ to write to a `/etc/sudoers.d/<username>` file and given the wide distro support for it maybe only support that approach. Debian documents that it'll ignore files ending in `~` or containing a `.` character so we should at least strip out any `.` characters from the file name.

Also, when we create the file with this approach you'll want to [set the permissions on it](https://doc.rust-lang.org/std/fs/struct.File.html#method.set_permissions) - you can probably just do something like the ssh provisioning: https://github.com/Azure/azure-init/blob/bc83b93e5f9d9638334012479b9876ccfa3fe4e6/libazureinit/src/provision/ssh.rs#L32

Reply:
Jeremy, I created a azure-init-user file under /etc/sudoers.d and added info for passwordless sudo for the user.